providers/implementations/encode_decode/ml_dsa_codecs.c

```diff
@@ -296,6 +305,11 @@ int ossl_ml_dsa_i2d_prvkey(const ML_DSA_KEY *key, uint8_t **out,
     const uint8_t *seed = ossl_ml_dsa_key_get_seed(key);
     const uint8_t *sk = ossl_ml_dsa_key_get_priv(key);
 
+    if (seed != NULL)
```

Comment:
perhaps nit...

why do we need `seed_only |= ossl_ml_dsa_key_get_prov_flags(....)` at line 309. I think the code here can be simplified to something like:
```
if (seed == NULL)
   seed_only = 0;
```